mono/metadata/w32socket-unix.c

```diff
@@ -320,11 +327,58 @@ mono_w32socket_recvfrom (SOCKET sock, char *buf, int len, int flags, struct sock
 
 	info = mono_thread_info_current ();
 
+	if (blocking) {
+		memset(&tstart, 0, timevallen);
+		memset(&tlimit, 0, timevallen);
+		memset(&timeout, 0, timevallen);
+
+		MONO_ENTER_GC_SAFE;
+		getsockopt (((MonoFDHandle*) sockethandle)->fd, SOL_SOCKET, SO_RCVTIMEO, &timeout, &timevallen);
+		MONO_EXIT_GC_SAFE;
+
+		if (timeout.tv_sec > 0 || timeout.tv_usec > 0) {
+			gettimeofday(&tstart, NULL);
```

Comment:
I'd prefer a monotonic clock (like [`mono_100ns_ticks`](https://github.com/mono/mono/blob/f6d04a53f962a57a136db267c105fe264f70b778/mono/utils/mono-time.h#L22-L23)) for this.